python/ray/autoscaler/_private/util.py

```diff
@@ -143,6 +144,10 @@ def prepare_config(config: Dict[str, Any]) -> Dict[str, Any]:
     - Has a valid Docker configuration if provided.
     - Has max_worker set for each node type.
     """
+    is_local = config["provider"]["type"] == "local"
```

Comment:
can this raise a `KeyError`?

Reply:
thanks for fixing -- looking forward to the day oss code uses pydantic instead of nested dicts